apps/dino-tracking/backend/src/main.py

```diff
@@ -0,0 +1,174 @@
+
+from core.annotations.outlines_overlay_node import OutlinesOverlayNode
+from core.object_selection.FE_prompt_services.click_prompt_service import (
+    ClickPromptService,
+)
+from core.annotations.dino_annotation_node import DinoAnnotationNode
+from core.encoder import Encoder
+from core.state_service import StateService
+from core.dino_similarity.similarity_heatmap_node import SimilarityHeatmapNode
+from core.object_selection.selection_mask_node import SelectionMaskNode
+
+load_dotenv(override=True)
+
+
+def main():
+    constants = YamlFilesLoader(Path(__file__).parent / "constants")
+    constants.load_all()
+
+    visualizer = dai.RemoteConnection(httpPort=8082)
+
+    device = dai.Device()
+    platform = device.getPlatformAsString()
+    print(f"Platform: {platform}")
+
+    with dai.Pipeline(device) as pipeline:
+        print("Creating pipeline...")
+
+        camera = pipeline.create(dai.node.Camera).build()
+
+        rgb_sensor = camera.requestOutput(
+            size=constants.camera.resolution,
+            type=dai.ImgFrame.Type.BGR888i,
+            fps=constants.camera.fps,
+        )
+
+        fastsam_rgb = camera.requestOutput(
+            size=constants.nn.segmentation.input_size,
+            type=dai.ImgFrame.Type.BGR888i,
+            fps=constants.camera.fps,
+        )
+
+        dino_rgb = camera.requestOutput(
+            size=constants.nn.dino.input_size,
+            type=dai.ImgFrame.Type.BGR888i,
+            fps=constants.camera.fps,
+        )
+
+        fastsam_nn = pipeline.create(ParsingNeuralNetwork).build(
+            input=fastsam_rgb, nn_source=constants.nn.segmentation.model_name
+        )
+        segmentation_out = fastsam_nn.out
+
+        dino_nn = pipeline.create(dai.node.NeuralNetwork).build(
+            input=dino_rgb,
+            nnArchive=dai.NNArchive(
+                dai.getModelFromZoo(
+                    dai.NNModelDescription(
+                        model=constants.nn.dino.model_name, platform=platform
+                    )
+                )
+            ),
+        )
+        dino_out = dino_nn.out
+
+        selection_node = pipeline.create(SelectionMaskNode).build(
+            frame_in=rgb_sensor,
+            segmentations=segmentation_out,
+        )
+
+        prompt_service = ClickPromptService(selection_node)
+        clear_service = ClearSelectionService(selection_node)
+        visualizer.registerService(prompt_service.NAME, prompt_service)
+        visualizer.registerService(clear_service.NAME, clear_service)
+
+        dino_grid = pipeline.create(DinoGridExtracorNode).build(dino_in=dino_out)
+
+        reference_node = pipeline.create(ReferenceFromSelectionNode).build(
+            mask_in=selection_node.out,
+            dino_in=dino_grid.out,
+            dino_input_size=constants.nn.dino.input_size,
+        )
+
+        adaptive_reference_node = pipeline.create(
+            AdaptiveReferenceVectorNode, constants.reference_adaptation
```

Comment:
inconsistent naming. Decide if its `reference_vector` or just `reference`. Imo better is `reference_vector` 
-->
`reference_vector_node`
`ReferenceVectorFromSelectionNode`
`adaptive_reference_vectore_node`
`AdaptiveReferenceVectorNode`